src/analytics/lambdas/custom-resource/get-source-prefix.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+
+import { CdkCustomResourceEvent, CdkCustomResourceResponse, CloudFormationCustomResourceEvent, Context } from 'aws-lambda';
+import { logger } from '../../../common/powertools';
+
+export interface ServiceTokenType {
+  readonly ServiceToken: string;
+}
+
+export interface GetResourcePrefixPropertiesType {
+  readonly odsEventPrefix: string;
+}
+
+type PropertiesType = ServiceTokenType & GetResourcePrefixPropertiesType;
+
+
+export const handler = async (event: CloudFormationCustomResourceEvent, context: Context) => {
+  logger.info('event', { event });
+  logger.info('context', { context });
+
+  const response: CdkCustomResourceResponse = {
+    Data: {
+      prefix: await _handler(event),
+    },
+    Status: 'SUCCESS',
+  };
+  logger.info('response', { response });
+  return response;
+};
+
+async function _handler(event: CdkCustomResourceEvent) {
+  const requestType = event.RequestType;
+  logger.info('RequestType: ' + requestType);
+
+  const odsEventPrefix = (event.ResourceProperties as PropertiesType).odsEventPrefix;
+
+  let prefix = odsEventPrefix;
+
+  if (odsEventPrefix.endsWith('/ods_events/') || odsEventPrefix.endsWith('/ods_events')) {
+    prefix = odsEventPrefix.replace(new RegExp('/ods_events[/]?'), '');
```

Comment:
Now you removed `ods_events` in prefix, what's the final prefix looking like?

Reply:
below are the prefixes for each table:

 - xxxx/project_id/ods_events/
 - xxxx/project_id/event/
 - xxxx/project_id/event_parameter/
 - xxxx/project_id/user/
 - xxxx/project_id/item/